docs/csharp/programming-guide/exceptions/snippets/exceptions/Program.cs

```diff
@@ -0,0 +1,240 @@
+            //<TryCatchStructure>
+            try
+            {
+                // Code to try goes here.
+            }
+            catch (SomeSpecificException ex)
+            {
+                // Code to handle the exception goes here.
+                // Only catch exceptions that you know how to handle.
+                // Never catch base class System.Exception without
+                // rethrowing it at the end of the catch block.
+            }
+            //</TryCatchStructure>
+
+            // <TryFinallyStructure>
+            try
+            {
+                // Code to try goes here.
+            }
+            finally
+            {
+                // Code to execute after the try block goes here.
+            }
+            // </TryFinallyStructure>
+
+            // <TryCatchFinallyStructure>
+            try
+            {
+                // Code to try goes here.
+            }
+            catch (SomeSpecificException ex)
+            {
+                // Code to handle the exception goes here.
+            }
+            finally
+            {
+                // Code to execute after the try (and possibly catch) blocks
+                // goes here.
+            }
+            // </TryCatchFinallyStructure>
+        }
+
+        // <ThrowMoreSpecificException>
+        int GetInt(int[] array, int index)
+        {
+            try
+            {
+                return array[index];
+            }
+            catch (IndexOutOfRangeException e)
+            {
+                throw new ArgumentOutOfRangeException(
+                    "Parameter index is out of range.", e);
+            }
+        }
+        // </ThrowMoreSpecificException>
+
+        private static void ReThrow()
+        {
+            // <RethrowError>
+            try
+            {
+                // Try to access a resource.
+            }
+            catch (UnauthorizedAccessException e)
+            {
+                // Call a custom error logging procedure.
+                LogError(e);
+                // Re-throw the error.
+                throw;
+            }
+            // </RethrowError>
+        }
+        static void LogError(Exception ex) { }
+
+
+        static void CloseOptionally()
+        {
+            //<CleanupIfNotNull>
+            FileStream? file = null;
+            FileInfo fileinfo = new System.IO.FileInfo("./file.txt");
+            try
+            {
+                file = fileinfo.OpenWrite();
+                file.WriteByte(0xF);
+            }
+            finally
+            {
+                // Check for null because OpenWrite might have failed.
+                file?.Close();
+            }
+            //</CleanupIfNotNull>
+        }
+
+        // <CantComplete>
+        static void CopyObject(SampleClass original)
+        {
+            if (original is null)
+            {
+                throw new ArgumentException("Parameter cannot be null", "original");
+            }
+        }
```

Comment:
```suggestion
        static void CopyObject(SampleClass original)
        {
            _ = original ?? throw new ArgumentException("Parameter cannot be null", nameof(original));
        }
```

Curious if this is better or not? One thing that you might want regardless of this style suggestion is the use of `nameof(original)`. Especially since this PR is about C# 6 and that is when I believe `nameof` was introduced. 😄